src/main/java/decodes/decoder/CsvFunction.java

```diff
@@ -53,12 +54,12 @@ public void execute(DataOperations dd, DecodedMessage decmsg) throws DecoderExce
 					trace("End of line seen.");
 					break;
 				}
-				if (c == ',')
+				if (dd.checkString(delimiter))
 				{
 					processColumn(col++, sb.toString().trim(), decmsg, dd.getCurrentLine(),
 						fieldStart, dd);
 					sb.setLength(0);
-					fieldStart = dd.getBytePos()+1;
+					fieldStart = dd.getBytePos()+delimiter.length();
```

Comment:
how does a "delimiter can be more than one character" work here? Are the multiple characters in the delimiter string treated as set of options the delimiter could be, or each field is delimited by all the listed characters?

Reply:
Basically you can require that the delimiter be a sequence of characters, or a word. E.g. if you use "bob" as the delimiter all the items are separated by the word "bob". I'll update the docs to make that more clear.

it's not done a lot but I've done it in the past for certain sets of data.